### PR TITLE
Add TOMAS-JAX as an in-process box model

### PR DESCRIPTION
**Pull request by aliakherati**
## Summary

Adds [TOMAS-JAX](https://github.com/reflective-org/tomas-jax) as a supported box model: a JAX implementation of the TwO-Moment Aerosol Sectional scheme, tracking number and per-species mass across 40 mass-doubling bins (~1.7 nm – 17.5 µm).

It differs from PartMC and MAM4 in one way that shapes the design: it's a Python/JAX library, not a compiled executable. There are no native input files to write and no command to invoke, so it's stepped **in process** rather than through `PoolRunner`.

Seven commits, each self-contained. Full details in [`TOMAS_JAX.md`](TOMAS_JAX.md), and
[`demo_tomas_jax.ipynb`](demo_tomas_jax.ipynb) is a runnable walkthrough with plots —
it needs no box-model binaries, so it runs straight after `pip install -r requirements.txt`.
It covers the size distribution evolving over 24 h, what each process contributes, and
TOMAS's sectional representation against the modal one MAM4 uses, built from the same
`Scenario`. Outputs are stripped, as elsewhere in the repo.

## Design

**No changes to shared framework code.** The adapter is one new module plus one line in `ambrs/__init__.py`; `aerosol_model.py`, `runners.py`, `analysis.py` and the existing adapters are untouched.

- `create_input` / `create_inputs` work exactly as for the other models.
- `run(input)` and `run_ensemble(inputs)` step the model in process, reusing one JIT-compiled step across a whole ensemble so compilation is paid once rather than per scenario.
- `retrieve_model_state(scenario_name, scenario, timestep, ...)` mirrors the PartMC and MAM4 functions, so callers can treat every model the same way. Since an in-process model leaves nothing on disk, it loads the state recorded by `write_input_files` and steps forward to `timestep`.
- `write_input_files` records the starting state (`.npz` + a readable `.txt`) purely for provenance and reproducibility.
- `invocation()` raises with an explanation rather than pretending there's an executable.

Each run returns the standard `analysis.Output` — a `part2pop` population plus gas mixture and thermodynamics — so `compute_variable`, `nmae` and `kl_divergence` work against it unchanged. There's a test asserting exactly that.

If you'd rather this went through a shared in-process runner (say a `DirectRunner` alongside `PoolRunner`), I'm happy to refactor — I kept the footprint minimal deliberately, but it's a reasonable thing to generalize.

## Modelling notes

- **Processes:** coagulation, condensation (H2SO4 → sulfate) and nucleation (`ricco_dunne` by default), applied in TOMAS's operator-split order.
- **Nucleation precursors** (`org_conc`, `nh3_conc`, `fion`) aren't carried by `Scenario`, so they're constructor settings; a `Scenario` NH3 concentration overrides `nh3_conc`.
- **Composition mapping is lossy and says so.** TOMAS carries sulfate, organics, ammonium and water. `BC`, `OIN`, `NO3`, `Cl`, `Na`, `Ca` and `CO3` have no slot; their mass is lumped into the organic slot so totals are conserved, and a warning names them. Full table in the docs.
- **Units:** `Scenario` modal number is `[# m^-3]` and TOMAS bins in `[# cm^-3]`; the adapter converts both ways. `gas_concs` are taken as `[molec cm^-3]`.
- **Two moments matter here.** A bin's mean particle mass is `Mk/Nk`, not the geometric bin mass, so output diameters are derived from `Mk/Nk`. That's what makes the population's mass match the model's — conserved to machine precision under coagulation, where using the geometric bin mass drifts by several percent. There's a regression test pinning this.

## Testing

25 new tests in `test/test_tomas_jax.py`, in the same `unittest.TestCase` style as the existing suite.

- **With tomas-jax installed:** `101 passed` (76 baseline + 25 new), 96% coverage of the new module.
- **Without it:** `76 passed, 25 skipped` — `import ambrs` still works and the tests skip themselves, so the dependency is genuinely optional at import time.

Verified with CI's own command, `coverage run --source=ambrs -m pytest`.

## Dependency note

tomas-jax is public, so I've declared it in `requirements.txt` over `git+https` and CI will exercise the new tests. It does pull in `jax`, `jaxlib` and `diffrax`. If you'd prefer not to make those mandatory for every install, say the word and I'll move it to an optional requirements file — the import is already guarded, so the tests skip cleanly either way.

Suggested labels: `box model integration`, `enhancement`, `python`, `dependencies`, `testing`.

### CODE REVIEW

**Comment by GlazerMann** [comment]
This adapter explicitly treats `scenario.gas_concs` as molecules/cm³:

```python
Gc[index] += _gas_conc_to_kg_per_cell(
    conc, self.boxvol, _GAS_MOLAR_MASS[gas.name])
```

but existing AMBRS usage points to a mol/mol-air convention:

* PartMC multiplies `scenario.gas_concs` by `1e9` when writing ppb.
* The existing demo explicitly labels `gas_concs` as `mol/mol-air`.
* The existing MAM4 adapter also treats them as gas mixing ratios.

Those interpretations differ by many orders of magnitude. At approximately 298 K and 1 atm, for example, `1e-9` mol/mol corresponds to roughly `2.5e10` molecules/cm³, not `1e-9` molecules/cm³.

Could AMBRS define one canonical `Scenario.gas_concs` representation and have TOMAS-JAX convert from it explicitly using the scenario temperature and pressure? A cross-adapter regression that starts PartMC and TOMAS-JAX from the same Scenario and verifies equivalent initial gas concentrations would be useful.

Two smaller integration issues are also worth considering:

1. `requirements.txt` installs TOMAS-JAX from moving branch `@dev`; pinning the exact tested commit would make installations reproducible.
2. `ambrs.__init__` eagerly imports `tomas_jax`, and TOMAS-JAX's config module executes `jax.config.update("jax_enable_x64", True)`. That means `import ambrs` can change process-wide JAX configuration. A lazy adapter import would avoid that optional model having a global import-time side effect.
